app/react/Library/components/specs/PermissionsFilter.spec.tsx

```diff
@@ -18,31 +21,84 @@ describe('Permissions Filter', () => {
     },
   };
 
-  const render = (aggs: Aggregations = aggregations) => {
-    component = shallow(<PermissionsFilter onChange={() => {}} aggregations={aggs} />);
+  const render = (
+    aggs: Aggregations = aggregations,
+    value: { level: string; refId: string }[] = []
+  ) => {
+    onChangeMock = jest.fn();
+    const props = {
+      value,
+      onChange: onChangeMock,
+      aggregations: aggs,
+    };
+
+    const state = {
+      user: fromJS({
+        _id: 'userId',
+        groups: [{ _id: 'groupId' }],
+      }),
+    };
+
+    component = renderConnected(PermissionsFilterUncontrolled, props, state);
   };
 
   it('should display the number of entities the user has permissions on', () => {
     render();
-    const options = component.find(SelectFilter).prop('options');
+    const options = component.find(MultiSelect).prop('options');
     expect(options).toEqual([
       expect.objectContaining({ value: 'write', results: 6 }),
       expect.objectContaining({ value: 'read', results: 3 }),
     ]);
   });
 
+  it('should map from level to ownIds-level pairs', () => {
+    render();
+    const onChange = component.find(MultiSelect).prop('onChange');
+    onChange(['read', 'write']);
+    expect(onChangeMock).toHaveBeenCalledWith([
+      { level: 'read', refId: 'userId' },
+      { level: 'read', refId: 'groupId' },
+      { level: 'write', refId: 'userId' },
+      { level: 'write', refId: 'groupId' },
+    ]);
+  });
+
+  it('should map from ownIds-level pairs to level', () => {
+    render(aggregations, []);
+    expect(component.find(MultiSelect).prop('value')).toEqual([]);
+
+    render(aggregations, [{ level: 'read', refId: 'userId' }]);
+    expect(component.find(MultiSelect).prop('value')).toEqual([]);
```

Comment:
Question for understading: why not ['read']?

Reply:
Already discussed this but for the record:
This mapping should only result in a read value in the array if every pair `(id, 'read')` with `id` in the set of the id's of the groups the user belongs to and the id of the user itself. That is: this filter is truly selecterd as `read` if the search is including every possible entity in which the user is assigned as an viewer either directly of through a group.